sdk/test-utils/recorder/src/sanitizer.ts

```diff
@@ -7,163 +7,177 @@ import { getRealAndFakePairs } from "./utils/connectionStringHelpers.js";
+    getSanitizerBodies(headerSanitizers, makeHeaderSanitizerBody),
+    getSanitizerBodies(uriSanitizers,
+      makeFindReplaceSanitizerBody("UriRegexSanitizer", "UriStringSanitizer")),
+    getSanitizerBodies(connectionStringSanitizers, makeConnectionStringSanitizerBody),
+    getSanitizerBodies(bodyKeySanitizers, makeBodyKeySanitizerBody),
+    getSanitizerBodies(continuationSanitizers, makeContinuationSanitizerBody),
+    (removeHeaderSanitizer
+      ? [
+        {
+          Name: "RemoveHeaderSanitizer",
+          Body: {
+            headersForRemoval: removeHeaderSanitizer.headersForRemoval.toString(),
+          },
+        },
+      ]
+      : []),
+    (oAuthResponseSanitizer ? [{ Name: "OAuthResponseSanitizer", Body: undefined }] : []),
+    [{ Name: "UriSubscriptionIdSanitizer", Body: uriSubscriptionIdSanitizer }],
+    (resetSanitizer ? [{ Name: "Reset", Body: undefined }] : []),
+  )
+}
 
 /**
- * Makes an AddSanitizer<boolean> function that adds the sanitizer if the value is set to true,
- * and otherwise makes no request to the server. Used for ResetSanitizer and OAuthResponseSanitizer.
+ * Makes an /addSanitizers request to the test proxy
  */
-const makeAddBodilessSanitizer =
-  (sanitizerName: ProxyToolSanitizers): AddSanitizer<boolean> =>
-  async (httpClient, url, recordingId, enable) => {
-    if (enable) {
-      await addSanitizer(httpClient, url, recordingId, {
-        sanitizer: sanitizerName,
-        body: undefined,
-      });
-    }
-  };
+export async function addSanitizers(
+  httpClient: HttpClient,
+  url: string,
+  recordingId: string | undefined,
+  options: SanitizerOptions,
+): Promise<void> {
+  const sanitizerBodies = makeBatchSanitizerBody(options);
+
+  if (sanitizerBodies.length === 0) return;
+  const uri = `${url}${paths.admin}${paths.addSanitizers}`;
+  const req = createRecordingRequest(uri, undefined, recordingId);
+
+  req.headers.set("Content-Type", "application/json");
+  req.body = JSON.stringify(sanitizerBodies);
+
+  logger.info("[addSanitizers] Adding sanitizers", options);
+  const rsp = await httpClient.sendRequest({
+    ...req,
+    allowInsecureConnection: true,
+  });
+  if (rsp.status !== 200) {
+    logger.error("[addSanitizers] addSanitizers request failed", rsp);
+    throw new RecorderError("addSanitizers request failed.");
+  }
+}
+
+// Batch sanitizer body makers
 
 /**
- * Makes an AddSanitizer function for a FindReplaceSanitizer, for example a bodySanitizer.
+ * Makes a sanitizer-bodies array from the sanitizers provided that is sent over as the addSanitizers request body.
+ * Also takes in the function that makes the body for each sanitizer.
+ */
+function getSanitizerBodies<T>(
+  sanitizers: T[] | undefined,
+  func: (sanitizer: T) => SanitizerRequestBody,
+): SanitizerRequestBody[] {
+  const bodies: SanitizerRequestBody[] = [];
+  if (!sanitizers) return bodies;
+  for (const sanitizer of sanitizers) {
+    bodies.push(func(sanitizer));
+  }
+  return bodies;
```

Comment:
nit: `return Array.from(sanitizers ?? [], (x) => func(x))`?

Reply:
this seems cool, will do that

Reply:
or `return sanitizers?.map(func) ?? []`?